jctools-core/src/main/java/org/jctools/queues/atomic/SpscUnboundedAtomicArrayQueue.java

```diff
@@ -14,47 +14,61 @@
 package org.jctools.queues.atomic;
 
 import org.jctools.util.Pow2;
-
+import java.util.concurrent.atomic.AtomicReferenceFieldUpdater;
+import java.util.concurrent.atomic.AtomicLongFieldUpdater;
+import org.jctools.queues.MessagePassingQueue;
+import org.jctools.queues.MessagePassingQueue.Supplier;
+import org.jctools.queues.MessagePassingQueueUtil;
+import org.jctools.queues.QueueProgressIndicators;
+import org.jctools.queues.IndexedQueueSizeUtil;
+import static org.jctools.queues.atomic.LinkedAtomicArrayQueueUtil.*;
 import java.util.concurrent.atomic.AtomicReferenceArray;
+import org.jctools.queues.MpmcArrayQueue;
 
+/**
+ * NOTE: This class was automatically generated by org.jctools.queues.atomic.JavaParsingAtomicLinkedQueueGenerator
+ * which can found in the jctools-build module. The original source file is SpscUnboundedArrayQueue.java.
+ */
 public class SpscUnboundedAtomicArrayQueue<E> extends BaseSpscLinkedAtomicArrayQueue<E> {
 
-    public SpscUnboundedAtomicArrayQueue(final int chunkSize) {
+    public SpscUnboundedAtomicArrayQueue(int chunkSize) {
         int chunkCapacity = Math.max(Pow2.roundToPowerOfTwo(chunkSize), 16);
         long mask = chunkCapacity - 1;
         AtomicReferenceArray<E> buffer = allocate(chunkCapacity + 1);
         producerBuffer = buffer;
         producerMask = mask;
         consumerBuffer = buffer;
         consumerMask = mask;
-        producerBufferLimit = mask - 1; // we know it's all empty to start with
-        soProducerIndex(0L);
```

Comment:
ATTENTION!: atomics had barrier this but unsafe version didn't. However this class nor it's parents write to a final field. This would have been an existing bug in the unsafe version, could you look this over and let me know your thoughts.